### PR TITLE
Add peg

### PR DESCRIPTION
**Pull request by satmandu**
Needed for Igt_gpu_tools PR.

Works properly:
- [x] x86_64

### CODE REVIEW

File: packages/peg.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+require 'package'
+
+class Peg < Package
+  description 'recursive-descent parser generators for C'
+  version '0.1.18-2'
+  source_url 'http://piumarta.com/software/peg/peg-0.1.18.tar.gz'
+  source_sha256 '20193bdd673fc7487a38937e297fff08aa73751b633a086ac28c3b34890f9084'
+
+  def self.build
+    system "make"
```
Comment (uberhacker) [inline packages/peg.rb:13]: Where is `CREW_PREFIX`?  If someone decides to change the prefix, how would they do it?

**Comment by satmandu** [comment]
There's no configure script. I should hack something like that in with sed.

File: packages/peg.rb (hunk 1 above)
Comment (satmandu) [reply packages/peg.rb]: Used sed to remove PREFIX and thus force use of PREFIX during make.

File: packages/peg.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+require 'package'
+
+class Peg < Package
+  description 'recursive-descent parser generators for C'
+  version '0.1.18-2'
+  source_url 'http://piumarta.com/software/peg/peg-0.1.18.tar.gz'
+  source_sha256 '20193bdd673fc7487a38937e297fff08aa73751b633a086ac28c3b34890f9084'
+
+  def self.build
+    system "sed -i 's/PREFIX  = \\/usr\\/local/PREFIX =/g' Makefile"
```
Comment (uberhacker) [inline packages/peg.rb:13]: `system "sed -i 's,PREFIX  = /usr/local,PREFIX = #{CREW_PREFIX},g' Makefile"`?

File: packages/peg.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    system "PREFIX=#{CREW_PREFIX} make"
+  end
+  def self.install
+    system "make ROOT=#{CREW_DEST_DIR} install"
```
Comment (uberhacker) [inline packages/peg.rb:17]: You could test with a different `PREFIX` than `/usr/local` but you might need `PREFIX=#{CREW_PREFIX}` here also.

File: packages/peg.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+class Peg < Package
+  description 'recursive-descent parser generators for C'
+  version '0.1.18-2'
+  source_url 'http://piumarta.com/software/peg/peg-0.1.18.tar.gz'
+  source_sha256 '20193bdd673fc7487a38937e297fff08aa73751b633a086ac28c3b34890f9084'
+
+  def self.build
+    system "sed -i 's/PREFIX  = \\/usr\\/local/PREFIX =/g' Makefile"
+    system "PREFIX=#{CREW_PREFIX} make"
```
Comment (uberhacker) [inline packages/peg.rb:14]: Actually, you might not need the sed command.  Try with a different `PREFIX` here and see what happens.

Comment (satmandu) [reply packages/peg.rb:14]: I was being overly anal about this. I'll delete the sed command entirely. It works fine with just the prefix before make.

File: packages/peg.rb (hunk 3 above)
Comment (satmandu) [reply packages/peg.rb:17]: Good point there.

File: packages/peg.rb (hunk 2 above)
Comment (satmandu) [reply packages/peg.rb]: this is perfect

**Comment by satmandu** [comment]
Turns out that without the sed on the makefile, it complains:

```
mv leg-new leg
Preconfiguring package...
cp -p peg /usr/local/bin/peg
strip /usr/local/bin/peg
cp -p leg /usr/local/bin/leg
strip /usr/local/bin/leg
Rename all *.la_tmp files back to *.la
Performing pre-install...
Performing install...
tar: ./usr/*: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
Performing post-install...
Peg installed!
```

**Pull request merged**